services/analyzer/app.py

```
@@ -49,109 +180,178 @@
 @app.on_event("shutdown")
 async def shutdown():
-    """Cleanup on shutdown."""
-    global redis_client, is_processing
-    is_processing = False
+    _shutdown_event.set()
     if redis_client:
-        await redis_client.close()
+        try:
+            await redis_client.close()
+        except Exception:
+            pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)